apis/cache/v1alpha1/cachebackend_types.go

```diff
@@ -0,0 +1,140 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	KubeDLCacheEnabled = "KubeDLCacheEnabled"
+)
+
+// CacheBackendSpec defines the desired state of CacheBackend
+type CacheBackendSpec struct {
+	// The location in the container where the dataset should be mounted
+	MountPath string `json:"mountPath,omitempty"`
+
+	// CacheEngine is different kinds of cache engine
+	CacheEngine *CacheEngine `json:"cacheEngine,omitempty"`
+}
+
+// CacheBackendStatus defines the observed state of CacheBackend
+type CacheBackendStatus struct {
+	// JobName indicates the job name of the cacheBackend binding
+	JobName string `json:"jobName,omitempty"`
+
+	// CacheStatus is used to help understand the status of a caching process
+	CacheStatus CacheStatus `json:"cacheStatus,omitempty"`
+}
+
+type CacheEngine struct {
+	// Fluid may be the only caching engine for the time being
+	Fluid *Fluid `json:"fluid,omitempty"`
+}
+
+type Fluid struct {
+	// Dataset describes the parameters related to the dataset file
+	// +required
+	Dataset *Dataset `json:"dataset,omitempty"`
+
+	// AlluxioRuntime is used to configure the cache runtime
+	// If this parameter is not specified, the cache will not take effect and the data set will be directly mounted
+	// +optional
+	AlluxioRuntime *AlluxioRuntime `json:"alluxioRuntime,omitempty"`
+}
+
+// Dataset is used to define where specific data sources are stored and mounted
+// A job may have multiple mount points, so it is a list
+type Dataset struct {
+	// DataSources is a list of dataset path
+	DataSources []DataSource `json:"dataSources,omitempty"`
+}
+
+type DataSource struct {
+	// The dataset path in variety file system
+	Path string `json:"path,omitempty"`
+
+	// SubdirectoryName is used as the file name of the data source in mountPath.
+	// The directory structure in container is as follows:
+	// - MountPath
+	// 	 - SubdirectoryName1
+	//   - SubdirectoryName2
+	//   - ...
+	SubdirectoryName string `json:"subdirName,omitempty"`
+}
+
+type AlluxioRuntime struct {
+	// Replicas is the min replicas of dataset in the cluster
+	Replicas int32 `json:"replicas,omitempty"`
+
+	// Fluid supports multi-tier configuration
+	TieredStorage []Level `json:"tieredStorage,omitempty"`
+}
+
+type Level struct {
+	// For Alluxio, it will use the cache path directory as its cache store
+	CachePath string `json:"cachePath,omitempty"`
+
+	// Quota defines a cache capacity of the directory
+	Quota string `json:"quota,omitempty"`
+
+	// Fluid sorts levels according to mediumType
+	MediumType string `json:"mediumType,omitempty"`
+}
+
+type CacheStatus string
+
+// The four CacheStatus vary as follows:
+// CacheCreating -> PVCCreating ( CacheSucceed ) -> PVCCreated
+// 			  or -> CacheFailed
+const (
+	PVCCreated    CacheStatus = "PVCCreated"
```

Comment:
does the PVCCreated mean the "cache loading" is done and ready for use ? otherwise, we may add additional status for that ?

Reply:
PVCCreated means that Fluid has bound the cache directory to the PVC. Generally, this PVC is ready for use.